riscv/src/continuations/bootloader.rs

```diff
@@ -223,38 +227,59 @@ and 3, 0, {PAGE_NUMBER_MASK}, 3;
 // Store & hash {WORDS_PER_PAGE} page words. This is an unrolled loop that for each each word:
 // - Loads the word into the P{{(i % 4) + 4}} register
 // - Stores the word at the address x3 * {PAGE_SIZE_BYTES} + i * {BYTES_PER_WORD}
-// - If i % 4 == 3: Hashes registers P0-P11, storing the result in P0-P3
+// - If i % 4 == 3: Hashes registers mem[0, 96), storing the result in mem[0, 32)
 //
-// At the end of the loop, we'll have a linear hash of the page in P0-P3, using a Merkle-Damgård
-// construction. The initial P0-P3 values are 0, and the capacity (P8-P11) is 0 throughout the
+// At the end of the loop, we'll have a linear hash of the page in [0, 32), using a Merkle-Damgård
+// construction. The initial [0, 32) values are 0, and the capacity [64, 96) is 0 throughout the
 // bootloader execution.
 
-{P0} <=X= 0;
-{P1} <=X= 0;
-{P2} <=X= 0;
-{P3} <=X= 0;
-{P4} <=X= 0;
-{P5} <=X= 0;
-{P6} <=X= 0;
-{P7} <=X= 0;
+// TODO currently the init call to poseidon_gl initializes the
+// scratch space with mstore_bootloader.
+// When the init call is removed, the calls below need to be replace by mstore_bootloader.
```

Comment:
Why would we remove it? We still need it because there might be 0 pages, right?

Reply:
I meant the poseidon_gl init call that's there because of witgen's dummy block

Reply:
Oh, right because we might be able to remove that once we support dynamically adding / skipping machines.